docker/docker-compose.yml

```diff
@@ -0,0 +1,35 @@
+version: "3"
+services:
+
+  clickhouse:
+    image: yandex/clickhouse-server
+    volumes:
+      - "./compose/graphite_rollup.xml:/etc/clickhouse-server/config.d/graphite_rollup.xml"
+      - "./compose/clickhouse_metrics.xml:/etc/clickhouse-server/config.d/clickhouse_metrics.xml"
+      - "./compose/init.sql:/docker-entrypoint-initdb.d/init.sql"
+      - "./data/clickhouse/data:/var/lib/clickhouse/data"
```

Comment:
When I just run `docker-compose up` it does create `repo_root/docker/data` as root with files from clickhouse inside and then the only root could delete it. I'm sure that it could be an issue for people without root access. I didn't find a quick fix for it, unfortunately =\

Reply:
clickhouse's docker image is root, I don't know how to solve this https://github.com/moby/moby/issues/2259

I can maybe comment out the volume mount to make it optional? Should I document the docker installation as well?

Reply:
Well, after https://github.com/yandex/ClickHouse/pull/4753 and the image update we could use the next config:

```diff
diff --git a/docker/docker-compose.yml b/docker/docker-compose.yml
index 044d46a..f320188 100644
--- a/docker/docker-compose.yml
+++ b/docker/docker-compose.yml
@@ -1,14 +1,19 @@
-version: "3"
+version: "3.2"
 services:
 
   clickhouse:
     image: yandex/clickhouse-server
+    user: "${UID:?You must do 'export UID' to launch}:${GID:?You must do 'export GID' to launch}"
     volumes:
       - "./compose/graphite_rollup.xml:/etc/clickhouse-server/config.d/graphite_rollup.xml"
       - "./compose/clickhouse_metrics.xml:/etc/clickhouse-server/config.d/clickhouse_metrics.xml"
       - "./compose/init.sql:/docker-entrypoint-initdb.d/init.sql"
-      - "./data/clickhouse/data:/var/lib/clickhouse/data"
-      - "./data/clickhouse/metadata:/var/lib/clickhouse/metadata"
+      - type: bind
+        source: ./data/clickhouse/
+        target: /var/lib/clickhouse/
+      - type: bind
+        source: ./data/clickhouse/
+        target: /var/log/clickhouse-server/
     ports:
       - "8123:8123"
 
```
Then just add `data/clickhouse/.gitkeep` and we are done. Or we could leave it for now and fix later

Reply:
So, @aksakali, the PR is merged and in the next `latest` docker image custom user will be implemented. I guess, we should wait for it

Reply:
Hey again, @aksakalli. Finally, the `latest` was built with the necessary changes.

Apply the next patch with `git am $filename`, please, and I'll merge the PR

```(diff)
From d1b7d5411402ac15b72c7a54ee39192dc2a9d4a8 Mon Sep 17 00:00:00 2001
From: "Mikhail f. Shiryaev" <mr.felixoid@gmail.com>
Date: Fri, 22 Mar 2019 10:59:37 +0100
Subject: [PATCH] Implement current_user in clickhouse container

---
 docker/data/clickhouse/.gitkeep | 0
 docker/docker-compose.yml       | 5 +++--
 2 files changed, 3 insertions(+), 2 deletions(-)
 create mode 100644 docker/data/clickhouse/.gitkeep

diff --git a/docker/data/clickhouse/.gitkeep b/docker/data/clickhouse/.gitkeep
new file mode 100644
index 0000000..e69de29
diff --git a/docker/docker-compose.yml b/docker/docker-compose.yml
index 044d46a..8e1d2ae 100644
--- a/docker/docker-compose.yml
+++ b/docker/docker-compose.yml
@@ -3,12 +3,13 @@ services:
 
   clickhouse:
     image: yandex/clickhouse-server
+    user: "${UID:?You must do 'export UID' to launch}:${GID:?You must do 'export GID' to launch}"
     volumes:
       - "./compose/graphite_rollup.xml:/etc/clickhouse-server/config.d/graphite_rollup.xml"
       - "./compose/clickhouse_metrics.xml:/etc/clickhouse-server/config.d/clickhouse_metrics.xml"
       - "./compose/init.sql:/docker-entrypoint-initdb.d/init.sql"
-      - "./data/clickhouse/data:/var/lib/clickhouse/data"
-      - "./data/clickhouse/metadata:/var/lib/clickhouse/metadata"
+      - "./data/clickhouse/:/var/lib/clickhouse/"
+      - "./data/clickhouse/:/var/log/clickhouse-server/"
     ports:
       - "8123:8123"
 
-- 
2.21.0

```

Reply:
I moved `docker/.gitignore` under `docker/data/clickhouse/` and changed to ignore everything except itself.